### ISSUE TITLE
MuseFX: Delay effect causes cut outs and repeated audio artefacts

### ISSUE THREAD

**Issue by Tantacrul** [musescore/MuseScore#14961]
**Describe the bug**
See the video 😄 

https://user-images.githubusercontent.com/21022311/205717451-ad73b575-a1e2-4410-833a-2778ae27dabc.mov

**Expected behavior**
Not this!

**Comment by bkunda** [comment]
whoa sh*t, YES THIS IS STILL HAPPENING.
Different for me though. 

WARNING: at 0:30, make sure your speaker volume is LOW. VERY LOW. And DO NOT attempt to use headphones:

https://user-images.githubusercontent.com/86290556/206202085-b56d691b-0266-4c68-9599-896f29abcd4b.mov

This effect seems to be doing all sorts of different things, EXCEPT produce a delay (random playback occurring without pressing play, notwithstanding).

**Comment by bkunda** [comment]
@davidwhearn here you go: 
[speed-test.mscz.zip](https://github.com/musescore/MuseScore/files/10193819/speed-test.mscz.zip)

**Comment by bkunda** [comment]
I suspect I didn't save the file after applying the delay effect. 

All I did was apply the delay effect to the grand piano track. The glitching sounds commenced straight away (without me having to initiate playback). I then started playback, and when I stopped playback the glitching continued: random bits of music would continue to play even without the score being in playback mode.

I've just had a look in my ...Library/Containers/Muse Hub//Data/InstallData/Effects file and I can see there are in fact two instances of the mac_Delay.vst3.zip file: one created 7 December 2022 and the other created 20 October 2022. Perhaps this is causing the problem?

**Comment by Tantacrul** [comment]
> 

It would be interesting to hear whether deleting the older effect helps @bkunda 

As for me, I've experienced the delay causing the same glitches in a recent build but not in another one (I've checked and can see that I do have the latest delay). It's maddeningly hard to reproduce.

**Comment by Tantacrul** [comment]
He he. Is there any way we can make this bug a feature?

**Comment by JoshuaPettus** [comment]
Indeed, MuseFX Delay will now turn your headphone users into Beethoven!

**Comment by bkunda** [comment]
@davidwhearn I've tried this and I suspect I haven't installed it correctly, because I'm still reproducing the old problem (and can't see any version number when I hover over the logo in the plugin UI).

I have added this .zip file (extracted) to my plugins file, and removed the old effect from the Library/Containers/MuseHub/InstallData/Effects file.

**Comment by Tantacrul** [comment]
I guess @bkunda: just reset to factory settings to be sure.

**Comment by bkunda** [comment]
Ok well MS seems to be looking in the Documents/MuseScore 4/Plugins folder for plugins, so this is where I put it. 
Just now, I didn't get any more of that horrible screeching sound, but I did still have notes randomly playing without my having initiated playback.

**Comment by bkunda** [comment]
Unfortunately I am not seeing the version number, so I suspect it remains incorrectly installed. 

I have removed the version from the Library/Containers folder, and I cannot locate another copy of this plugin on my machine, save for in the Documents/MuseScore4/Plugins folder, which is where MuseScore tells me it is looking to find plugins: 

<img width="638" alt="Screenshot 2022-12-13 at 1 51 24 pm" src="https://user-images.githubusercontent.com/86290556/207322900-f8ae55e2-de01-40c6-82b3-241ef9958dfb.png">
